src/core.cpp

```diff
@@ -3,55 +3,112 @@
 #include <random>
 
 #include "core.h"
+#include "action.h"
 
-double Character::GetMoney() {
-    return this->money_;
+RelationshipModifier::RelationshipModifier(Character* target, int start_date, int end_date, double opinion_modifier_) {
+    this->target_ = target;
+    this->start_date_ = start_date;
+    this->end_date_ = end_date;
+    this->opinion_modifier_ = opinion_modifier_;
 }
 
-void Character::SetMoney(double money) {
+Character::Character(int id, double money) {
+    this->id_ = id;
     this->money_ = money;
 }
 
-
-Action::Action(Character* actor, double score) {
-    this->actor_ = actor;
-    this->score_ = score;
+int Character::GetId() const {
+    return this->id_;
 }
 
-Action::~Action() {
-}
-
-Character* Action::GetActor() {
-    return this->actor_;
+double Character::GetMoney() const {
+    return this->money_;
 }
 
-void Action::SetActor(Character* actor) {
-    this->actor_ = actor;
+void Character::SetMoney(double money) {
+    this->money_ = money;
 }
 
-double Action::GetScore() {
-    return this->score_;
+void Character::ExpireRelationships(int now) {
+    for (auto it = this->relationships_.begin(); it != this->relationships_.end();){
+        RelationshipModifier* relationship = it->get();
+        if(now >= relationship->end_date_) {
+            printf("expiring %d -> %d (%f)\n", this->GetId(), relationship->target_->GetId(), relationship->opinion_modifier_);
+            this->relationships_.erase(it++);
```

Comment:
This is invalid, because vector::erase invalidates iterators at or after erased. Consider using `it = relationships_.erase(it);` instead.

Reply:
I pulled this from a SO post that said this would work, (and it seems to in practice). but it's convenient that erase returns the next element, which is what I _think_ I'm trying to do here :)